README.md

```diff
@@ -21,23 +22,56 @@ An example application showing the use OpenWeave on the Nordic nRF52840.
 
 ## Introduction
 
+![nrf52840 DK](doc/images/nrf52840-dk.jpg)
+
 The nRF52840 lock example application provides a working demonstration of a connected door lock device, built using OpenWeave, OpenThread and the Nordic
 nRF5 SDK. The example supports remote access and control of a simulated door lock over a low-power, 802.15.4 Thread network.  It is capable of being paired into
 an existing Weave network along with other Weave-enabled devices, and supports communication (via a Weave border router device) with the Nest service.    The
 example targets the [Nordic nRF52840 DK](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-DK) development kit, but is readily 
-adaptable to other hardware. 
+adaptable to other nRF52840-based hardware. 
 
-The nRF52840 lock example is intended to serve both as a means to explore the workings of OpenWeave and OpenThread, as well as a template for creating real
+The lock example is intended to serve both as a means to explore the workings of OpenWeave and OpenThread, as well as a template for creating real
 products based on the Nordic platform. 
 
 The example application builds upon the [OpenWeave](https://github.com/openweave/openweave-core/) and [OpenThread](https://github.com/openthread/openthread)
 projects, which are incorporated as submodules and built from source.  A top-level Makefile orchestrates the entire build process, including building OpenWeave,
 OpenThread and select files from the nRF5 SDK.  The resultant image file  can be flashed directly onto the Nordic dev kit hardware.
 
 
-<a name="app-behavior"></a>
+<a name="software-architecture"></a>
+
+## Software Architecture
```

Comment:
I haven't cross-checked; however, to the extent it's possible, we should probably align any of our descriptive language here with similar context at openweave.io for any of these topic areas.